File: src/retin/toolbox/document/list/StringList.cpp
/**
 * \file StringList.cpp
 * \author Philippe H. Gosselin
 * \version 4.0
 */

#include "StringList.h"

#include "../serialization/Serializers.h"

RETIN_SERIALIZABLE_OBJECT_DEFINITIONS(StringList);

namespace retin {

bool	StringList::equals(const Document* doc) const 
{
	if (!doc || !doc->isInstanceOf<StringList>())
		return false; 
	const StringList* list = static_cast<const StringList*>(doc);
	if (list->vData.size() != vData.size())
		return false;
	for (uint i=0;i<vData.size();i++)
		if (vData[i] != list->vData[i])
			return false;
	return true;
}


}

File: src/retin/toolbox/document/list/StringList.java
/**
 * \file StringList.java
 * \author Philippe H. Gosselin
 * \version 4.0
 */

package retin.toolbox.document.list;

import java.util.ArrayList;
import retin.toolbox.document.Document;

public class StringList extends Document
{
	private ArrayList<String> strings;

	public	StringList ()
	{
		strings = new ArrayList<String>();
	}
        
        public	StringList (ArrayList<String> strings)
	{
		this.strings = strings;
	}

	public	int	size() { return strings.size(); }
	public	String	get(int i) { return strings.get(i); }
	public	void	add(String s) { strings.add(s); }
}
